src/main/java/org/testeditor/fixture/web/WebDriverFixture.java

```diff
@@ -145,10 +154,89 @@ public WebDriver startBrowser(String browser) {
 		}
 		configureDriver();
 		return driver;
-    }
+	}
+
+	public void initWithReporter(TestRunReporter reporter) {
+		reporter.addListener(this);
+	}
+
+	@Override
+	public void reported(SemanticUnit unit, Action action, String msg) {
+		if (unit == SemanticUnit.TEST && action == Action.ENTER) {
+			runningTest = msg;
+		}
+		if (screenshotShouldBeMade(unit, action, msg)) {
+			screenshot(msg + '.' + action.name());
+		}
+	}
+
+	private String runningTest = null;
+	private static final int SCREENSHOT_FILENAME_MAXLEN = 128;
+
+	private String getCurrentTestCase() {
+		return runningTest != null ? runningTest : "UNKNOWN_TEST";
+	}
+
+	private String getScreenshotPath() {
+		// configurable through maven build?
+		return "screenshots";
+	}
+
+	private boolean screenshotShouldBeMade(SemanticUnit unit, Action action, String msg) {
+		// configurable through maven build?
+		return ((action == Action.ENTER) || unit == SemanticUnit.TEST) && driver != null;
+	}
+
+	private String reduceToMaxLen(String base, int maxLen) {
+		if (base.length() < maxLen) {
+			return base;
+		} else {
+			return base.substring(0, maxLen);
+		}
+	}
+
+	private String constructScreenshotFilename(String filenameBase, String testcase) {
+		String additionalGraphicType = ".png";
+		String escapedBaseName = filenameBase.replaceAll("[^a-zA-Z0-9.-]", "_").replaceAll("_+", "_")
+				.replaceAll("_+\\.", ".").replaceAll("\\._+", ".");
+		String timeStr = new SimpleDateFormat("HHmmss.SSS").format(new Date());
+		StringBuffer finalFilenameBuffer = new StringBuffer();
+		int lenOfFixedElements = timeStr.length() + additionalGraphicType.length() + 1/* hyphen */;
+		finalFilenameBuffer //
```

Comment:
SimpleDateFormat only with Hour, Minute, Seconds and Milliseconds could be more clearly when also a date is specified for detailed description of tests